README.md

```diff
@@ -1 +1,106 @@
-# cairo-annotations
+# Cairo Annotations
+
+The `cairo-annotations` crate provides tools tailored for working with annotations in the Cairo language.
+These
+annotations are part of the Sierra Debug Information format and serve to enrich the code by detailing information.
+
+> 📝 **Note**
+>
+> Although this crate is primarily utilized by projects like `cairo-coverage`, `cairo-profiler`, and `starknet-foundry`,
+> it is also fully capable of functioning as a standalone library to work with Sierra Debug Information annotations.
+
+## Features of Cairo Annotations
+
+### Structured Annotations
+
+`cairo-annotations` offers a structured representation, allowing for more ergonomic manipulation of annotations.
+Some
+key features include:
+
+- **Coverage Annotations**: Track locations in the Cairo code that correspond to specific Sierra statements.
+- **Profiler Annotations**: Provide mappings from Sierra statements to fully qualified Cairo paths, detailing which
+  functions in the Cairo code triggered them.
+
+All annotations implement the `TryFromDebugInfo` trait, enabling their extraction from Sierra debug information.
+Here's
+a simple example:
+
+```rust
+let annotations = VersionedCoverageAnnotations::try_from_debug_info(sierra_debug_info).unwrap();
```

Comment:
it may be a good idea to show what `annotation` will look like? not sure though

Reply:
You mean json form?

Reply:
yes, but probably with heavily mocked data. or at least some indication on return type or something

Reply:
Added

Reply:
perfect, thanks. Hopefully this won't be a pain to maintain 🙏